pkg/daemon/cleanup/cleanup_test.go

```diff
@@ -0,0 +1,355 @@
+					"test2",
+					mgr.Config{
+						BinaryPathName: "test2 arg1 arg2",
+						Dependencies:   []string{},
+						Description:    windows.ManagedTag + " test2",
+					},
+					svc.Status{State: svc.Running},
+				),
+				"test3-unmanaged": fake.NewFakeService(
+					"test3-unmanaged",
+					mgr.Config{
+						BinaryPathName: "test3-unmanaged",
+						Dependencies:   []string{},
+					},
+					svc.Status{State: svc.Running},
+				),
+			},
+			expectedRemovedServices:   []string{"test1", "test2", "test3-unmanaged"},
+			expectedRemainingServices: []string{},
+			preserveNode:              false,
+			servicesCMExists:          true,
+			configMapServices: []servicescm.Service{
+				{
+					Name:         "test1",
+					Command:      "test1 arg1",
+					Dependencies: []string{"test2", "test3-unmanaged"},
+					Bootstrap:    false,
+					Priority:     3,
+				},
+				{
+					Name:         "test2",
+					Command:      "test2 arg1 arg2",
+					Dependencies: []string{},
+					Bootstrap:    false,
+					Priority:     2,
+				},
+				{
+					Name:         "test3-unmanaged",
+					Command:      "test3-unmanaged",
+					Dependencies: nil,
+					Bootstrap:    false,
+					Priority:     0,
+				},
+			},
+		},
+		// TODO: Tests without desired services ConfigMap in the cluster (best effort cleanup)
+	}
+
+	nodeIP := "192.168.10.10"
+	testAddrs := []net.Addr{&net.IPNet{IP: net.ParseIP(nodeIP)}}
+
+	for _, test := range testIO {
+		t.Run(test.name, func(t *testing.T) {
+			desiredVersion := "testversion"
+			// This ConfigMap's name must match with the given Node object's desired-version annotation
+			cm, err := servicescm.Generate(servicescm.NamePrefix+desiredVersion,
+				"openshift-windows-machine-config-operator", &servicescm.Data{test.configMapServices,
+					[]servicescm.FileInfo{}})
+			require.NoError(t, err)
+			clusterObjs := []client.Object{
+				// This is the node object that will be used in these test cases
+				&core.Node{
+					ObjectMeta: meta.ObjectMeta{
+						Name: "node",
+						Annotations: map[string]string{
+							nodeconfig.DesiredVersionAnnotation: desiredVersion,
+						},
+					},
+					Status: core.NodeStatus{
+						Addresses: []core.NodeAddress{{Address: nodeIP}},
+					},
+				},
+			}
+			if test.servicesCMExists {
+				clusterObjs = append(clusterObjs, cm)
+			}
+			fakeClient := clientfake.NewClientBuilder().WithObjects(clusterObjs...).Build()
+
+			winSvcMgr := fake.NewTestMgr(test.existingServices)
+			err = Deconfigure(fakeClient, testAddrs, winSvcMgr, test.preserveNode)
+			require.NoError(t, err)
+			allServices, err := winSvcMgr.GetServices()
+			require.NoError(t, err)
+
+			for _, svc := range test.expectedRemovedServices {
+				_, present := allServices[svc]
+				assert.Falsef(t, present, "service %s not removed as expected", svc)
+			}
+			for _, svc := range test.expectedRemainingServices {
+				_, present := allServices[svc]
+				assert.Truef(t, present, "service %s not present as expected", svc)
+			}
```

Comment:
Instead of doing this it makes more sense to check the final state of the services.
Instead of having two slices `expectedRemovedServices` and `expectedRemainingServices`, you just have
`expectedServices` and assert that the service list is equal to it